Allen & Heath Xone K2/K1 mapping part 2

Be-ing opened the pull request:
I realized I could fit in the `loop_in` and `loop_out` Controls for manual loops with the supershift mode. This works well because these Controls don't require showing any state on the LEDs.

Be-ing committed:
remove outdated comment

Be-ing committed:
A&H Xone K2/K1: map loop_in and loop_out with supershift mode

loop_in & loop_out do not require showing any state on the LEDs, so
they work well for supershift mode.

Be-ing committed:
A&H Xone K2/K1: add mode to control all parameters of 1 effect unit

accessed by pressing the Layer button of the 4 effect unit layouts

Be-ing commented:
I have added a new mode that allows for controlling the parameters of all the effects in one unit at a time. It is only available in the 4 effect unit layouts and is activated with the Layer button in the bottom left. Documentation is on the [wiki](https://mixxx.org/wiki/doku.php/allen_heath_xone_k2#effect_unit_focusing).

ferranpujolcamins commented on res/controllers/Xone-K2-scripts.js:
```diff
@@ -424,6 +427,9 @@ XoneK2.Deck = function (column, deckNumber, midiChannel) {
         },
+        supershift: function () {
+            this.inKey = 'loop_out';
+        },
```
This is pending to be documented in the wiki, right?

Be-ing replied:
yes

Be-ing replied:
documented

ferranlala reviewed (changes requested):
Your new mode is nice! However, I miss the possibility to change effects without having to exit the focused mode.

Be-ing commented:
> I miss the possibility to change effects without having to exit the focused mode.

I thought about that but I am not sure where to fit it into the design. Maybe shift + the first parameter knob?

ferranpujolcamins commented:
You could map each parameter knob

Be-ing commented:
That happens to make the code a little simpler, so I'll do that.

Be-ing commented:
Now that I have implemented this new mode, I am really looking forward to a GSOC student implementing parameter rearrangement this summer. It would be cool to make that somehow mappable for controller, for example maybe a new ControlObject to cycle the last parameter to the first parameter position. I'll have to think about this more...

Be-ing committed:
Components: fix bug with effect selection for some controllers

... that do not have a way to query the controller on startup for the
position of all the knobs.

Be-ing committed:
A&H Xone K2/K1: use shift + knob turn for effect selection

when a single effect unit is focused. Any of the knobs for the effect
parameters can be used.

Be-ing committed:
A&H Xone K2/K1: map master and headphones effect routing buttons

These replace the deck 3 & 4 buttons for 2 deck layouts. For the 4
effect unit layouts, the master and headphone routing buttons are
accessible with shift.

Be-ing commented:
Okay, I implemented effect switching by turning any parameter knob with shift in the effect unit focused mode.

I changed the 2 deck + 2 effect unit layouts so the bottom two effect routing buttons control the routing for the master and headphone channels. I don't think there's a point to having those mapped for routing to decks 3 and 4 when those decks are not mapped on the controller. Also, with the 4 effect unit layout, the deck 3 & 4 routing buttons switch to controlling the master and headphones channels with shift and turn amber instead of red.

I'm thinking about switching the shifted and unshifted functions of the bottom two amber buttons for the decks in loop mode. That is, they'd do beatjumping normally and loop double/halve with shift. What do you think about that?

ferranpujolcamins commented:
> I'm thinking about switching the shifted and unshifted functions of the bottom two amber buttons for the decks in loop mode. That is, they'd do beatjumping normally and loop double/halve with shift. What do you think about that?

I don't have a strong preference about that. What you prefer will be ok :)

About your last changes, I won't be able to test them soon.

Be-ing committed:
A&H Xone K2/K1: rename files to match conventions

Be-ing committed:
A&H Xone K2/K1: swap beatjumping and loop halve/double functions

In my experience accessing beatjumping quickly is more important
than quickly adjusting the loop size.

Be-ing commented:
@ferranpujolcamins can you do one last check to make sure nothing is obviously messed up?

Be-ing commented:
I just played a set with this for a few hours. I am quite happy with the decision to make beatjumping the default action of the bottom buttons instead of loop halve/double. It is very useful on a controller without jog wheels.
